observatory-platform/observatory/platform/utils/test_utils.py

```diff
@@ -166,13 +171,84 @@ def save_empty_file(path: str, file_name: str) -> str:
     return file_path
 
 
+def make_prefix(test_name: str, org_name: str):
+    """Creates a unique prefix for buckets and datasets for the unit tests using
+    the capital letters of both the test and organisation name. 
+    
+    :param test_name: Name of the unit test.
+    :param org_name: Organisation name.
+    :return prefix: Prefix for the buckets and datasets.
+    """
+
+    if re.search("[A-Z]+", org_name):
+         
+        prefix = (
+        ("".join(re.findall("[A-Z]+", test_name))).lower()
+        + "_"
+        + ("".join(re.findall("[A-Z]+", org_name))).lower()
+        )
+
+    # no capitals in the org_part and "_" separating the name. Used for function names.
+    elif not re.search("[A-Z]+", org_name) and re.search("[\_]+", org_name):
+
+        reg = re.compile(r"(?:(?<=\_)|^)(?:[a-z]|\d+)", re.I)
+        org_part = ''.join(reg.findall(org_name))
+
+        prefix = (
+        ("".join(re.findall("[A-Z]+", test_name))).lower()
+        + "_"
+        + org_part
+        )
+
+    elif org_name != "":
+
+        # For cases where the organisation name is too long and will cause 
+        # the bucket name length to be greater than the max of 63 characters.
+         
+        if len(org_name) <= 10:
+            org_part = org_name.strip().replace(" ","_").replace("-","_")
+
+        else:
+            if re.search("[\-]+", org_name):
+                org_split = org_name.split("-")
+                
+                org_part = ""
+                for part_name in org_split:
+                    org_part = org_part + part_name[0]
+            elif re.search("[\s]+", org_name):
+                org_split = org_name.split(" ")
+                
+                org_part = ""
+                for part_name in org_split:
+                    org_part = org_part + part_name[0]
+            else:
+                org_part = org_name[0]
+        
+        prefix = (
+        ("".join(re.findall("[A-Z]+", test_name))).lower()
+        + "_"
+        + org_part
+        )
+
+    else:
+
+        prefix = (
+        ("".join(re.findall("[A-Z]+", test_name))).lower()
+        )
+
+    logging.info(f"Prefix for buckets and datsets: {prefix}")
+
+    return prefix
+
+
 class ObservatoryEnvironment:
```

Comment:
- It would be good if you didn't have to specify the prefix when creating `ObservatoryEnvironment`. To make it simple we could have a standard global prefix, defined as a keyword argument in the constructor, e.g. "obs_tests". It is OK if one test deletes another test's datasets or buckets, as long as enough time has passed to know that they are not still being used.

- We should probably also modify `add_bucket` to have an optional prefix parameter like `add_dataset` so that the test writer can make buckets more easily identified, we just need to be careful / have checks because of the bucket character limit.

- We could call `delete_old_test_datasets` and `delete_old_test_buckets` from within the `create` function, in the finally statement, after line 645, then the datasets and buckets will always be removed and we don't need to add extra code to all existing unit tests (probably just the tests added in test_gc_utils) :

```
    if self.create_gcp_env:
        # Remove Google Cloud Storage buckets
        for bucket_id in self.buckets:
            self._delete_bucket(bucket_id)
        # Remove BigQuery datasets
        for dataset_id in self.datasets:
            self._delete_dataset(dataset_id)

       # Call delete_old_test_datasets and delete_old_test_buckets here
```

* Lastly, the above change means that we could put the `age_to_delete` variable as a keyword argument in the constructor of `ObservatoryEnvironment`. Perhaps set as 1 or 2 hours?